ddtrace/propagation/http.py

```diff
@@ -197,11 +210,17 @@ def _inject(span_context, headers):
     @staticmethod
     def _extract(headers):
         # type: (Dict[str, str]) -> Optional[Context]
-        trace_id = _extract_header_value(
-            POSSIBLE_HTTP_HEADER_TRACE_IDS,
-            headers,
+        trace_id = int(
+            _extract_header_value(
+                POSSIBLE_HTTP_HEADER_TRACE_IDS,
+                headers,
+            )
+            or 0
         )
-        if trace_id is None:
+        if trace_id == 0 or trace_id > _MAX_UINT_64BITS:
+            log.debug(
+                "Invalid trace id: %d. `x-datadog-trace-id` must be greater than zero and less than 2**64 - 1", trace_id
```

Comment:
is `2**64 - 1` the max allowed?

does the message need to say "and less than 2**64" ? (since 2**64 - 1 should be inclusive)

Reply:
Yup, I'll fix this 2**64 - 1 is a valid traced id (since trace ids are unsigned)